### PR TITLE
Tooling: Setup Redux/Thunk for global state management. https://trell…

### PR DESCRIPTION
**Pull request by w3bh4ck**
…o.com/c/TvRoWRMU/48-tooling-setup-redux-thunk-for-global-state-management

### CODE REVIEW

**Comment by tomijaga** [comment]
@w3bh4ck What do you think of using redux toolkit so that we can avoid most of the boilerplate that comes with using just redux.

File: src/redux/action-types/transactions.types.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const transactionTypes = {
+  GET_ALL_TRANSACTIONA: 'GET_ALL_TRANSACTIONA',
```
Comment (tomijaga) [inline src/redux/action-types/transactions.types.ts:2]: @w3bh4ck Should this be `Get_All_Transactions` instead

File: src/redux/reducers/transactions.reducer.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+};
+
+interface Actions {
+  type: string;
+  payload: string;
+}
+
+export default (state = initialState, action: Actions) => {
+  switch (action.type) {
+    case transactionTypes.GET_ALL_TRANSACTIONA:
```
Comment (tomijaga) [inline src/redux/reducers/transactions.reducer.ts:14]: Here to

File: src/redux/action-types/transactions.types.ts (hunk 1 above)
Comment (w3bh4ck) [reply src/redux/action-types/transactions.types.ts:2]: @tomijaga yea, that was a typo... I'll correct that now

**Comment by w3bh4ck** [comment]
> @w3bh4ck What do you think of redux toolkit instead of plain redux so that we can avoid most of the boilerplate.
> 
> https://redux.js.org/redux-toolkit/overview



> @w3bh4ck What do you think of redux toolkit instead of plain redux so that we can avoid most of the boilerplate.
> 
> https://redux.js.org/redux-toolkit/overview

@tomijaga yea you are right, my thought initially was going with something that any new contributor will be familiar with. But you can still make an update so that we can switch to Redux-toolkit as long as everyone is familiar with it.

**Comment by tomijaga** [comment]
I merged your previous PR and it created conflicts. Can you fix it please?

**Comment by tomijaga** [comment]
> @tomijaga yea you are right, my thought initially was going with something that any new contributor will be familiar with. But you can still make an update so that we can switch to Redux-toolkit as long as everyone is familiar with it.

@w3bh4ck  Are you familiar with it?
I have seen it used before and read the documentation but haven't used it in a project myself.

@nishp77 is the other person on the team and he is said he is willing to learn it

**Pull request closed without merging**